src/differentials/composite.jl

```diff
@@ -230,6 +241,7 @@ function elementwise_add(a::NamedTuple{an}, b::NamedTuple{bn}) where {an, bn}
     end
 end
 
+elementwise_add(a::Dict, b::Dict) = merge(+, a, b)
```

Comment:
Can we move this defintion to be with [the definitions in `src/differential_arithmetic.jl`](https://github.com/JuliaDiff/ChainRulesCore.jl/blob/84493b32cdf1087a32b4a4918c4172593b5c2784/src/differentials/composite.jl#L184-L186)?